corehq/apps/domain/views.py

```diff
@@ -1951,9 +1950,9 @@ def post(self, request, *args, **kwargs):
             elif request.POST.get('old_documentation_file', False):
                 new_domain.documentation_file_path = old.documentation_file_path
                 new_domain.documentation_file_type = old.documentation_file_type
-            new_domain.save()
 
             if publish_on_submit:
+                new_domain.save()
                 _publish_snapshot(request, self.domain_object, published_snapshot=new_domain)
```

Comment:
I like the explicit save here, and there are code paths in publish snapshot where it will not get saved so this maintains past behavior better, but I do not have strong opinions on this.